### PR TITLE
add dataset info

### PR DESCRIPTION
**Pull request by jin-s13**
It requires mmcv-full (latest version). https://github.com/open-mmlab/mmcv/pull/1083
Blocked by mmcv release.

===================================================================================
1. Add one `dataset_info` file for each dataset, e.g. "dataset_info/atrw.py". This file contains all information about the dataset: (1) `dataset_name` (2)`paper_info` (3)`keypoint_info` (4)`skeleton_info`. 
2. Replace `dataset` with `dataset_info` in `inference.py`.
3. Change 1-based indexing to 0-based indexing for `skeleton`.
4. pose_limb_color -> pose_link_color
5. add `_base_` datasets in `mmpose/datasets/datasets/`

### CODE REVIEW

File: dataset_info/atrw.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+dataset_info = dict(
+    paper_info=dict(
```
Comment (innerlee) [inline dataset_info/atrw.py:2]: Add homepage?

File: dataset_info/atrw.py [hunk 2]
```diff
@@ -98,49 +98,44 @@
...
-            id=14,
-            color=[51, 153, 255],
-            type='lower',
-            swap=None),
+            name='center', id=14, color=[51, 153, 255], type='lower', swap=''),
     },
     skeleton_info={
         0:
-        dict(pair=('left_ear', 'nose'), id=0, color=[0, 255, 0]),
+        dict(limb=('left_ear', 'nose'), id=0, color=[0, 255, 0]),
```
Comment (innerlee) [inline dataset_info/atrw.py:108]: Reason for this change?

File: configs/_base_/datasets/300w.py [hunk 3]
```diff
@@ -382,3 +382,23 @@
     skeleton_info={},
     joint_weights=[1.] * 68,
     sigmas=[])
+
+data_root = 'data/300w'
```
Comment (innerlee) [inline configs/_base_/datasets/300w.py:386]: These do not belong to the definition of the dataset.

**Comment by innerlee** [comment]
Another comment: the `_base_` name in source code feels not right

image

**Comment by jin-s13** [comment]
@cherryjm Could you please help add `configs/_base_/datasets/mpi_inf_3dhp.py`?

**Comment by jin-s13** [comment]
@zengwang430521 Could you help check the color in `configs/_base_/datasets/interhand3d.py`, which is used for visualization?

**Comment by innerlee** [comment]
Change `_base_` folder to `base` here

image

File: mmpose/datasets/datasets/animal/animal_base_dataset.py [hunk 4]
```diff
@@ -1,148 +0,0 @@
...
-
-import json_tricks as json
-import numpy as np
-from torch.utils.data import Dataset
-from xtcocotools.coco import COCO
-
-from mmpose.datasets.pipelines import Compose
-
-
-class AnimalBaseDataset(Dataset, metaclass=ABCMeta):
```
Comment (innerlee) [inline mmpose/datasets/datasets/animal/animal_base_dataset.py:12]: Keep a dummy class of `AnimalBaseDataset` here for compatibility

File: mmpose/models/detectors/interhand_3d.py [hunk 5]
```diff
@@ -108,7 +108,7 @@ def show_result(self,
                     bbox_color='green',
                     thickness=2,
                     pose_kpt_color=None,
-                    pose_limb_color=None,
+                    pose_link_color=None,
```
Comment (innerlee) [inline mmpose/models/detectors/interhand_3d.py:111]: Compatibility for this change.

**Pull request merged**